### PR TITLE
fixed dereferences after null check

### PR DESCRIPTION
**Pull request by wargio**
* 1372641 Dereference after null check
* 1372644 Dereference after null check

### CODE REVIEW

File: libr/io/section.c [hunk 1]
```diff
@@ -16,12 +16,14 @@ static void section_free(void *p) {
 }
 
 R_API void r_io_section_init(RIO *io) {
-	if (io && !io->sections) {
-		if (!(io->sections = ls_newf (section_free))) {
-			return;
+	if (io) {
+		if(!io->sections){
```
Comment (radare) [inline libr/io/section.c]: Missing spaces

File: libr/io/section.c [hunk 2]
```diff
@@ -16,12 +16,14 @@ static void section_free(void *p) {
...
+			if (!(io->sections = ls_newf (section_free))) {
+				return;
+			}
 		}
```
Comment (radare) [inline libr/io/section.c]: Use a constant. Not fffffff

File: libr/io/p_cache.c [hunk 3]
```diff
@@ -367,11 +367,14 @@ static bool __desc_fini_cb (void *user, void *data, ut32 id) {
 }
 
 R_API void r_io_desc_cache_fini(RIODesc *desc) {
+	__desc_fini_cb (NULL, (void *) desc, 0);
+/*
```
Comment (XVilka) [inline libr/io/p_cache.c]: Just remove this code, no need to keep it.

Comment (wargio) [inline libr/io/p_cache.c]: done.

**Pull request merged**